src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -661,19 +741,28 @@ namespace
         const RenderType        type,
         const bool              use_max_proc_maps,
         const TimeValue         time,
+        RendProgressCallback*   progress_cb,
         ObjectMap&              object_map,
-        MaterialMap&            material_map)
+        MaterialMap&            material_map,
+        AssemblyMap&            assembly_map)
     {
-        for (const auto& object : entities.m_objects)
+        for (size_t i = 0, e = entities.m_objects.size(); i < e; ++i)
         {
+            const auto& object = entities.m_objects[i];
             add_object(
                 assembly,
                 object,
                 type,
                 use_max_proc_maps,
                 time,
                 object_map,
-                material_map);
+                material_map,
+                assembly_map);
+
+            const int done = static_cast<int>(i);
+            const int total = static_cast<int>(e);
+            if (progress_cb->Progress(done, total) == RENDPROG_ABORT)
```

Comment:
Note that `done` will at most reach `total - 1`, never `total`.

Reply:
Does it make sense to do this: `progress_cb->Progress(done + 1, total)`?
So that very first iteration will already be 1, not zero?

Reply:
Yes it's correct since after the first iteration you already accomplished one unit of work.